src/scripts/main.js

```diff
@@ -1,5 +1,35 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  if (e.target.closest('.wall')) {
+    const wallRect = wall.getBoundingClientRect();
+    const spiderWidth = spider.offsetWidth;
+    const spiderHeight = spider.offsetHeight;
+
+    const wallStyle = getComputedStyle(wall);
+    const wallBorderRght = parseInt(wallStyle.borderRightWidth);
+    const wallBorderBottom = parseInt(wallStyle.borderBottomWidth);
+
+    let xCoords = e.clientX - wallRect.left;
+    let yCoords = e.clientY - wallRect.top;
+
+    xCoords = Math.max(
+      0,
+      Math.min(
+        xCoords - spiderWidth / 2,
+        wallRect.width - spiderWidth - wallBorderRght * 2,
```

Comment:
The calculation for `xCoords` should consider only one instance of `wallBorderRght` instead of multiplying it by 2. The border is applied only once on the right side, so using `wallBorderRght` is sufficient.